src/gui/wizard/selftest.cpp

```diff
@@ -5,107 +5,65 @@
 #include "selftest_MINI.h"
 #include "marlin_client.h"
 #include "DialogHandler.hpp"
+#include "eeprom.h"
 
-StateFncData StateFnc_SELFTEST_FAN(StateFncData last_run) {
+WizardState_t StateFnc_SELFTEST_FAN() {
     marlin_test_start(stmFans);
     DialogHandler::WaitUntilClosed(ClientFSM::SelftestFans, 0);
-    return last_run.PassToNext();
+    return WizardState_t::next;
 }
 
-StateFncData StateFnc_SELFTEST_X(StateFncData last_run) {
+WizardState_t StateFnc_SELFTEST_X() {
     marlin_test_start(stmXAxis);
     DialogHandler::WaitUntilClosed(ClientFSM::SelftestAxis, 0);
-    return last_run.PassToNext();
+    return WizardState_t::next;
 }
 
-StateFncData StateFnc_SELFTEST_Y(StateFncData last_run) {
+WizardState_t StateFnc_SELFTEST_Y() {
     marlin_test_start(stmYAxis);
     DialogHandler::WaitUntilClosed(ClientFSM::SelftestAxis, 0);
-    return last_run.PassToNext();
+    return WizardState_t::next;
 }
 
-StateFncData StateFnc_SELFTEST_Z(StateFncData last_run) {
+WizardState_t StateFnc_SELFTEST_Z() {
     marlin_test_start(stmZAxis);
     DialogHandler::WaitUntilClosed(ClientFSM::SelftestAxis, 0);
-    return last_run.PassToNext();
+    return WizardState_t::next;
 }
 
-StateFncData StateFnc_SELFTEST_XYZ(StateFncData last_run) {
+WizardState_t StateFnc_SELFTEST_XYZ() {
     marlin_test_start(stmXYZAxis);
     DialogHandler::WaitUntilClosed(ClientFSM::SelftestAxis, 0);
-    return last_run.PassToNext();
+    return WizardState_t::next;
 }
 
-StateFncData StateFnc_SELFTEST_PASS(StateFncData last_run) {
-    const char en_text[] = N_("All tests finished successfully!");
-    string_view_utf8 translatedText = _(en_text);
-    MsgBoxPepa(translatedText, Responses_Next);
-    return last_run.PassToNext().PassToNext(); // 2x PassToNext() to skip fail
-}
-
-StateFncData StateFnc_SELFTEST_FAIL(StateFncData last_run) {
-    static const char en_text[] = N_("The selftest failed to finish. Double-check the printer's wiring and axes. Then restart the Selftest.");
-    string_view_utf8 translatedText = _(en_text);
-    MsgBox(translatedText, Responses_Next);
-    return StateFncData(WizardState_t::EXIT, WizardTestState_t::PASSED);
+WizardState_t StateFnc_SELFTEST_RESULT() {
+    static const char en_text_ok[] = N_("All tests finished successfully!");
+    static const char en_text_nok[] = N_("The selftest failed to finish. Double-check the printer's wiring and axes. Then restart the Selftest.");
```

Comment:
This message is too generic and not very helpful. I get this message all the time on my MINI (I think it's because it doesn't cool off fast enough) and it's not useful to the user or customer support. It will be better to add an error code that specifies *what* failed during the self test so the users can search for a specific problem or they can pass that information to customer service which can diagnose where the problem is.  There are multiple steps during the self test, we should specify why it failed.

Reply:
@ftrofin thank your for your comment - you are right, this is something we'd like to improve further - now the Wizard can distinguish what failed and report accordingly. Stay tuned.